craft_parts/xattrs.py

```diff
@@ -0,0 +1,84 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2019-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers to read and write filesystem extended attributes."""
+
+import os
+import sys
+from typing import Optional
+
+from craft_parts import errors
+
+_STAGE_PACKAGE_KEY = "origin_stage_package"
```

Comment:
this might be a separations of concern leak. Maybe add a TODO to improve this handling.